mikelangelo-project/unik
# Openstack integr basic

# Purpose or Intent

This PR introduces OpenStack provider. It consists of three commits:
- vendor code only
- stubs for OpenStack only
- implementation of basic functions

Please consult commit messages for more details.
# Testing
1. start DevStack and obtain it's IP address
2. add openstack provider to the daemon-config.yml, as described [here](https://github.com/mikelangelo-project/unik/blob/openstack-integr-basic/docs/providers/openstack.md)
3. run daemon
4. `cd` into this folder: 
   [Example-JavaBurekMinimal.zip](https://github.com/mikelangelo-project/unik/files/442427/Example-JavaBurekMinimal.zip)
   and run to STAGE IMAGE + RUN INSTANCE :

```
./unik_openstack_up.sh
```

and then to DELETE IMAGE and DELETE INSTANCE:

```
./unik_openstack_down.sh
```

After the first command an image and instance should appear on your OpenStack - instanc should print "BUREK" to the console 10 times.
